app/services/appointment_notification_service.rb

```diff
@@ -16,17 +16,32 @@ def initialize(appointments:, days_overdue: 3)
 
   def send_after_missed_visit
     eligible_appointments = appointments.eligible_for_reminders(days_overdue: days_overdue)
+    next_messaging_time = Communication.next_messaging_time
 
     eligible_appointments.each do |appointment|
+      # i don't believe this is the best way to control this. combined with the fact that
+      # we grab all appointments with remind_on before today, I suspect this means we're
+      # resending the same reminders because they failed the first time.
+      # leaving it in place for now in an effort to change as little of the current process as possible.
       next if appointment.previously_communicated_via?(communication_type)
 
-      next_messaging_time = Communication.next_messaging_time
+      appointment_reminder = create_appointment_reminder(appointment)
 
-      AppointmentNotification::Worker.perform_at(next_messaging_time, appointment.id, communication_type)
+      AppointmentNotification::Worker.perform_at(next_messaging_time, appointment_reminder.id, communication_type)
     end
   end
 
   private
 
+  def create_appointment_reminder(appointment)
+    AppointmentReminder.create!(
+      appointment: appointment,
+      patient: appointment.patient,
+      remind_on: appointment.remind_on,
+      status: "pending",
+      message: "sms.appointment_reminders.#{communication_type}"
```

Comment:
Does the `sms` prefix make sense in this key anymore?  Because this also could be for WhatsApp right?

Reply:
Agreed. That is true for "TwilioSmsDeliveryDetail" too. And, if we really want to make changes here, I don't think there are any differences between any of the sms and whatsapp translations. I think it was a premature optimization.